deepchecks/checks/performance/class_performance_imbalance.py

```diff
@@ -0,0 +1,239 @@
+
+
+class ClassPerformanceImbalanceCheck(SingleDatasetBaseCheck):
+    """Visualize class imbalance by displaying the difference between class metrics.
+
+    Args:
+        alternative_metrics (Mapping[str, Union[str, Callable]]):
+            An optional dictionary of metric name or scorer functions
+
+    Raises:
+        ValueError:
+            if provided dict of metrics is emtpy;
+            if one of the entries of the provided metrics dict contains name of unknown scorer;
+            if one of the entries of the provided metrics dict contains not callable value;
+    """
+
+    def __init__(
+        self,
+        alternative_metrics: t.Optional[t.Mapping[str, AlternativeMetric]] = None
+    ):
+        super().__init__()
+        self.alternative_metrics: t.Optional[t.Mapping[str, MetricFunc]] = None
+
+        if alternative_metrics is not None and len(alternative_metrics) == 0:
+            raise ValueError('alternative_metrics - expected to receive not empty dict of scorers!')
+
+        elif alternative_metrics is not None:
+            self.alternative_metrics = {}
+
+            for name, metric in alternative_metrics.items():
+                if isinstance(metric, t.Callable):
+                    self.alternative_metrics[name] = metric
+                elif isinstance(metric, str):
+                    self.alternative_metrics[name] = get_scorer(metric)
+                else:
+                    raise ValueError(
+                        f"alternative_metrics - expected to receive 'Mapping[str, Callable]' but got " #pylint: disable=inconsistent-quotes
+                        f"'Mapping[str, {type(metric).__name__}]'!" #pylint: disable=inconsistent-quotes
+                    )
+
+    def run(
+        self,
+        dataset: Dataset,
+        model: t.Any # TODO: find more precise type for model
+    ) -> CheckResult:
+        """Run Check.
+
+        Args:
+            dataset (Dataset): a dataset object
+            model (BaseEstimator): A scikit-learn-compatible fitted estimator instance
+
+        Returns:
+            CheckResult: value is a dictionary of labels, value of each key is another dictionary
+                with calculated metrics
+
+        Raises:
+            DeepchecksValueError:
+                if task type is not binary or multi-class;
+                if dataset does not have a label column or the label dtype is unsupported;
+                if provided dataset is empty;
+        """
+        return self._class_performance_imbalance(dataset, model)
+
+    def _class_performance_imbalance(
+        self,
+        dataset: Dataset,
+        model: t.Any # TODO: find more precise type for model
+    ) -> CheckResult:
+        check_name = type(self).__name__
+        expected_model_types = [ModelType.BINARY, ModelType.MULTICLASS]
+
+        dataset.validate_label(check_name)
+        dataset.validate_features(check_name)
+        Dataset.validate_dataset(dataset, check_name)
```

Comment:
Need to be first, if it's not Dataset object, the previous calls will fail on unwanted exception

Reply:
updated